### PR TITLE
Fix cythonization of preproc

### PR DESCRIPTION
**Pull request by kif**
Close to #1349
related to https://github.com/cython/cython/issues/3560

### CODE REVIEW

**Comment by kif** [comment]
I will double check the other cython files as well

File: pyFAI/ext/histogram.pyx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+#cython: boundscheck=False, wraparound=False, cdivision=True, initializedcheck=False,
+## This is for developping
+## cython: profile=True, warn.undeclared=True, warn.unused=True, warn.unused_result=False, warn.unused_arg=True
 #
 #    Project: Fast Azimuthal integration
 #             https://github.com/silx-kit/pyFAI
 #
 #
-#    Copyright (C) 2012-2019 European Synchrotron Radiation Facility, Grenoble, France
+#    Copyright (C) 2012-20120 European Synchrotron Radiation Facility, Grenoble, France
```
Comment (scoder) [inline pyFAI/ext/histogram.pyx:11]: typo, I guess.

File: pyFAI/ext/preproc.pyx [hunk 2]
```diff
@@ -62,7 +63,7 @@ cdef floating[::1]c1_preproc(floating[::1] data,
                              floating delta_dummy=0.0,
                              bint check_dummy=False,
                              floating normalization_factor=1.0,
-                             ) nogil:
+                             ) with gil:
```
Comment (scoder) [inline pyFAI/ext/preproc.pyx:66]: I think the GIL is not even released when these functions are called below, so you can just leave out the GIL declaration completely now (unless it's called elsewhere as well).

**Comment by kif** [comment]
Hi Stefan,

Thanks for having alook at my code.

On Wed, 29 Apr 2020 04:24:52 -0700
Stefan Behnel <notifications@github.com> wrote:

> @scoder commented on this pull request.
> 
> 
> 
> >  #  
>  #    Project: Fast Azimuthal integration
>  #             https://github.com/silx-kit/pyFAI
>  #
>  #
> -#    Copyright (C) 2012-2019 European Synchrotron Radiation Facility, Grenoble, France
> +#    Copyright (C) 2012-20120 European Synchrotron Radiation Facility, Grenoble, France
> 
> typo, I guess.

Indeed

> 
> > -                             ) nogil:  
> +                             ) with gil:
> 
> I think the GIL is not even released when these functions are called below, so you can just leave out the GIL declaration completely now (unless it's called elsewhere as well).

Actually those functions are called in plenty of other modules some of them are maybe already in nogil-state

Cheers,

Jerome

File: pyFAI/ext/splitBBoxCSR.pyx [hunk 3]
```diff
@@ -260,7 +260,7 @@ class HistoBBox1d(object):
             position_t[::1] cpos0_sup = self.cpos0_sup, cpos0_inf = self.cpos0_inf, cpos1_min, cpos1_max,
             mask_t[::1] cmask
             acc_t inv_area, delta_left, delta_right
-            cnumpy.int64_t memsize, key_page_cnt, key_page_size, lut_nbytes
+            long memsize, key_page_cnt, key_page_size, lut_nbytes
```
Comment (scoder) [inline pyFAI/ext/splitBBoxCSR.pyx:263]: The best type for sizes is `Py_ssize_t`

File: pyFAI/ext/morphology.pyx [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
+#cython: embedsignature=True, language_level=3
+#cython: boundscheck=False, wraparound=False, cdivision=True, initializedcheck=False,
+## This is for developping:
+##cython: profile=True, warn.undeclared=True, warn.unused=True, warn.unused_result=False, warn.unused_arg=True
 #
 #    Project: Fast Azimuthal integration
 #             https://github.com/silx-kit/pyFAI
 #
-#    Copyright (C) 2014-2018 European Synchrotron Radiation Facility, Grenoble, France
+#    Copyright (C) 2014-20120European Synchrotron Radiation Facility, Grenoble, France
```
Comment (scoder) [inline pyFAI/ext/morphology.pyx:10]: Another typo here.

**Comment by kif** [comment]
> -            cnumpy.int64_t memsize, key_page_cnt, key_page_size, lut_nbytes
> +            long memsize, key_page_cnt, key_page_size, lut_nbytes
> 
> The best type for sizes is `Py_ssize_t`

Thanks, I learned something.

**Pull request merged**